docs/operation/ros_interface/ros2.rst

```diff
@@ -62,30 +64,86 @@ Parameters
   * **Type**: bool
-  * **Description**: Setting this to true publishes the odom->base_link TF following REP 105.
+  * **Description**: Setting this to true publishes the odom->base_link TF
+    following REP 105.
   * **Default**: ``false``
 
 * ``odom_frame_name``:
 
   * **Type**: string
-  * **Description**: The name of the odometry frame when publishing the odom->base_link TF.
+  * **Description**: The name of the odometry frame when publishing the
+    odom->base_link TF.
   * **Default**: ``"odom"``
 
 * ``base_frame_name``:
 
   * **Type**: string
-  * **Description**: The name of the base_link frame when publishing the odom->base_link TF.
+  * **Description**: The name of the base_link frame when publishing the
+    odom->base_link TF.
   * **Default**: ``"base_link"``
 
-Usage
------
+Examples
+--------
+
+The **slate_base_node** is designed to be controlled by other user-written nodes that publish to the ``cmd_vel`` topic or call the services provided by the **slate_base_node**. Below are some examples of how to interact with the **slate_base_node** using ROS 2 command line tools.
+
+Setting Parameters
+^^^^^^^^^^^^^^^^^^
+
+You can set parameters for the **slate_base_node** when running it using the ``--ros-args`` flag.
+
+.. code-block:: bash
+
+  # Set the update frequency to 20 Hz
+  $ ros2 run interbotix_slate_driver slate_base_node --ros-args -p update_frequency:=20
+
+  # Enable publishing of the odom->base_link TF
+  $ ros2 run interbotix_slate_driver slate_base_node --ros-args -p publish_tf:=true
+
+  # Set the odometry frame name to "my_odom"
+  $ ros2 run interbotix_slate_driver slate_base_node --ros-args -p odom_frame_name:="my_odom"
 
-Using ros2run
-^^^^^^^^^^^^^
+  # Set the base frame name to "my_base"
+  $ ros2 run interbotix_slate_driver slate_base_node --ros-args -p base_frame_name:="my_base"
+
+Calling Services
+^^^^^^^^^^^^^^^^
+
+You can call services provided by the **slate_base_node** using the ``ros2 service call`` command.
 
 .. code-block:: bash
 
-  $ ros2 run interbotix_slate_driver slate_base_node
+  # Disable charging
+  $ ros2 service call /enable_charging std_srvs/srv/SetBool "{data: false}"
+
+  # Set the lights to color BLUE
+  $ ros2 service call /set_light_state interbotix_slate_msgs/srv/SetLightState "{light_state: 4}"
+
+  # Set the text to "hello world" on the SLATE screen
+  $ ros2 service call /set_text interbotix_slate_msgs/srv/SetString "{data: 'hello world'}"
+
+  # Disable torque on the drive wheels
+  $ ros2 service call /set_motor_torque_status std_srvs/srv/SetBool "{data: false}"
+
+Publishing Velocity Commands
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can publish velocity commands to the **slate_base_node** using the ``ros2 topic pub`` command. ``-r 10`` sets the publishing rate to 10 Hz.
+
+.. code-block:: bash
+
+  # Publish a velocity command to move forward
+  $ ros2 topic pub -r 10 /cmd_vel geometry_msgs/msg/Twist "{linear: {x: 0.1, y: 0.0, z: 0.0}, angular: {x: 0.0, y: 0.0, z: 0.0}}"
```

Comment:
ros2 topic pub has an option to publish a specified number of times: `-t TIMES, --times TIMES`. Use this with -r to have it publish for, say, 3 seconds. I don't think this option is available with `rostopic pub` as far as I can tell